src/main/java/de/tum/cit/aet/artemis/programming/domain/ProgrammingExerciseBuildConfig.java

```diff
@@ -292,6 +300,112 @@ public void setSolutionCheckoutPath(String solutionCheckoutPath) {
         this.solutionCheckoutPath = solutionCheckoutPath;
     }
 
+    /**
```

Comment:
This adds a lot of domain-logic that (currently?) is specific to LocalCI. I do not think this belongs into the database entity object that should only provide ‘dumb’ getters/setters for its attributes and smaller utilities that ensure the object invariants are upheld.

I think it would be better to move the added code to a new utility class, or a service class instead.